webpack/components/extensions/HostDetails/Cards/ContentViewDetailsCard.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  Card,
+  CardHeader,
+  CardTitle,
+  CardBody,
+  Flex,
+  FlexItem,
+  GridItem,
+  Label,
+} from '@patternfly/react-core';
+
+import Skeleton from 'react-loading-skeleton';
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { propsToCamelCase } from 'foremanReact/common/helpers';
+import PropTypes from 'prop-types';
+import ContentViewIcon from '../../../../scenes/ContentViews/components/ContentViewIcon';
+
+const HostContentViewDetails = ({
+  contentView, lifecycleEnvironment, contentViewVersionId,
+  contentViewVersion, contentViewVersionLatest,
+}) => {
+  let versionLabel = `Version ${contentViewVersion}`;
+  if (contentViewVersionLatest) {
+    versionLabel += ' (latest)';
+  }
+
+  return (
+    <GridItem xl2={3} md={6} lg={5}>
+      <Card isHoverable>
+        <CardHeader>
+          <CardTitle>{__('Content View Details')}</CardTitle>
+        </CardHeader>
+        <CardBody>
+          <DescriptionList isAutoColumnWidths>
+            <DescriptionListGroup>
+              <DescriptionListTerm>{__('Content View')}</DescriptionListTerm>
+              <DescriptionListDescription>
+                <Flex>
+                  <FlexItem spacer={{ default: 'spacerNone' }}><ContentViewIcon composite={contentView.composite} /></FlexItem>
+                  <FlexItem><a href={`/content_views/${contentView.id}`}>{`${contentView.name}`}</a> </FlexItem>
+                  <FlexItem><Label color="purple" href={`/lifecycle_environments/${lifecycleEnvironment.id}`}>{`${lifecycleEnvironment.name}`}</Label></FlexItem>
+                </Flex>
+              </DescriptionListDescription>
+            </DescriptionListGroup>
+            <DescriptionListGroup>
+              <DescriptionListTerm>{__('Version in use')}</DescriptionListTerm>
+              <DescriptionListDescription>
+                <Flex>
+                  <FlexItem>
+                    <a href={urlBuilder(`content_views/${contentView.id}/versions/${contentViewVersionId}`, '')}>
+                      {versionLabel}
+                    </a>
+                  </FlexItem>
+                </Flex>
+              </DescriptionListDescription>
+            </DescriptionListGroup>
+          </DescriptionList>
+        </CardBody>
+      </Card>
+    </GridItem>
+  );
+};
+
+const ContentViewDetailsCard = ({ hostDetails }) => {
+  if (hostDetails.content_facet_attributes) {
+    return <HostContentViewDetails {...propsToCamelCase(hostDetails.content_facet_attributes)} />;
+  }
+  return <Skeleton />;
```

Comment:
Even with the Patternfly skeleton this doesn't render anything useful.  I think we should move the `GridItem`, `Card`, `CardHeader`, and `CardBody` here, to the outer `ContentViewDetailsCard` component. And then just conditionally render either the `Skeleton` or the `DescriptionList`.

(And after you refactor all that, enjoy not having to change your React testing library test ;)

Reply:
Hmm don't agree on moving the grid item and card here. Currently the card does not show up if you don't pass the `content-facet-attributes`.  If we set grid item we'll be seeing empty card which we don't want.

Reply:
We do want an empty card! Otherwise it renders a skeleton on the page and not in the card.

Reply:
If you don't want anything to render at all, you wouldn't need a Skeleton

Reply:
I basically need the equivalent of "dont render anything". If `Skeleton` doesn't do that we need something else.

Reply:
```js
return null;
```
would do that. And I'm fine with that in this case

Reply:
Skeleton lets the user know that data is still loading, so that makes sense to me here as well. But I'm fine either way

Reply:
Oh I should ve just returned null looks like.

Reply:
updated!